course/viewbase.py

```diff
@@ -74,20 +76,25 @@ def get_resource_objects(self):
             # Try to find a language that is defined for this course instance
             # and apply it
             if self.instance.language:
+                instance_languages = self.instance.language.strip('|').split('|')
+                instance_def_language = instance_languages[0]
+                instance_languages = set(instance_languages)
 
                 languages = []
+                query_language = self.request.GET.get('hl')
+                if query_language:
+                    if query_language[:2] in instance_languages:
+                        languages.append(query_language)
+                    else:
+                        raise TranslationNotFound
                 if self.user_course_data and self.user_course_data.language:
```

Comment:
Yes, the part where we look for enrollment and profile languages could be skipped, the last part should be still included because it checks that the course is available in the (query) language.

Reply:
Isn't that done already above? `if query_language[:2] in instance_languages:`

Reply:
Oh that is true, don't know how I didn't notice that.